### PR TITLE
feat: Add hasActivity param to derived addresses

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
check for has activity for derived addresses

Important changes:
- [x] Something worth noting for reviewers.

Closes #

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?
- [ ] Have you tested changes with mobile?
- [x] Have you tested changes with desktop?

File: node/status_node_services.go [hunk 1]
```diff
@@ -441,9 +441,9 @@ func (b *StatusNode) appmetricsService() common.StatusService {
 	return b.appMetricsSrvc
 }
 
-func (b *StatusNode) walletService(accountsFeed *event.Feed, openseaAPIKey string) common.StatusService {
+func (b *StatusNode) walletService(accountsFeed *event.Feed, openseaAPIKey string, accountsDB *accounts.Database) common.StatusService {
 	if b.walletSrvc == nil {
-		b.walletSrvc = wallet.NewService(b.appDB, b.rpcClient, accountsFeed, openseaAPIKey)
+		b.walletSrvc = wallet.NewService(b.appDB, b.rpcClient, accountsFeed, openseaAPIKey, b.gethAccountManager, accountsDB)
```
Comment (alaibe) [inline node/status_node_services.go]: nit: appDB is already a parameter, you don t really need to add accountsDB which can be rebuild from appDB?

File: services/wallet/api.go [hunk 2]
```diff
@@ -293,3 +304,74 @@ func (api *API) GetSuggestedFees(ctx context.Context, chainID uint64) (*Suggeste
...
+		alreadyExists := false
+		for _, account := range addedAccounts {
+			if types.Address(common.HexToAddress(info[derivedPath].Address)) == account.Address {
+				alreadyExists = true
+				break
+			}
+		}
+
+		var ctx context.Context
+		transactions, err := api.s.transferController.GetTransfersByAddress(ctx, api.s.rpcClient.UpstreamChainID, common.HexToAddress(info[derivedPath].Address), nil, (*hexutil.Big)(big.NewInt(1)), false)
```
Comment (alaibe) [inline services/wallet/api.go]: I think this loop and in particular this call should be made in a coroutine to execute all the calls in parallel. If that helps, you can see how it is done here: https://github.com/status-im/status-go/blob/develop/services/wallet/token.go#L245

File: node/status_node_services.go (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply node/status_node_services.go]: aaah ok let me check :)

File: services/wallet/api.go (hunk 2 above)
Comment (Khushboo-dev-cpp) [reply services/wallet/api.go]: hey thanks for pointing to an example. It was just really quick when I tested it and didnt do it. Also do you mean https://github.com/status-im/status-go/blob/d884749ebf5e4edff46295c27e583425820e64ac/services/wallet/token.go#L260? cause this qpi getTokenBalance is just a normal function right?

Comment (Khushboo-dev-cpp) [reply services/wallet/api.go]: hey made getting each of the derived address concurrent ! @alaibe

**Review by alaibe — APPROVED** [review]

**Comment by churik** [comment]
@Khushboo-dev-cpp any clues on what can be affected on mobile-client side? 
Thanks!

**Comment by Khushboo-dev-cpp** [comment]
> @Khushboo-dev-cpp any clues on what can be affected on mobile-client side? Thanks!

hey @churik this only adds new Apis for account generation and giving the user to choose a derivation path and corresponding address. Doesnt touch any already implemented API.

**Review by qoqobolo — APPROVED** [review]

**Comment by qoqobolo** [comment]
@Khushboo-dev-cpp thanks for your patience and waiting! PR is tested on the Mobile side.

**Pull request merged**